lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1064,6 +1064,43 @@ public function get_stylesheet( $type = 'all' ) {
 		}
 	}
 
+	/**
+	 * Returns a filtered tree by the given base and property.
+	 * It keeps the version intact.
+	 *
+	 * @param string $base One of the top-level keys.
+	 * @param array $property The property to filter the object by.
+	 *
+	 * @return WP_Theme_JSON
+	 */
+	public function filter_by( $base = '', $property = array() ) {
```

Comment:
By absorbing this logic we make consumers a bit more decoupled from path changes. It also absorbs complex logic such as: when a consumer does `$tree->filter_by( 'settings', array( 'color', 'palette' ) )` we should return all the color palettes (the global and the per-block ones).

In principle, having this applied to any top-level key seems reasonable but I'm also open to create a `filter_settings_by` method instead.